samples/nrf9160/modem_shell/src/uart/uart_shell.c

```diff
@@ -77,10 +106,9 @@ void uart_toggle_power_state_at_event(const struct lte_lc_evt *const evt)
 {
 	if (evt->type == LTE_LC_EVT_MODEM_SLEEP_ENTER) {
 		mosh_print("Modem sleep enter: disabling UARTs requested");
-		k_sleep(K_MSEC(500)); /* allow little time for printing the notification */
-		disable_uarts();
+		k_work_submit(&uart_disable_work_q);
```

Comment:
Was the sleep removed on purpose? Did you test that it works without? Just wondering because similar sleep is still present above in `cmd_uart_disable()`.

Reply:
The work queue introduces some delay, but I can't guarantee that it is enough. Doesn't `mosh_print` use blocking mode?

It was @balaji-nordic that implemented this fix, and he's on vacation now. I can test it, but it's not clear to me what commands I need to send to trigger this event. Can you please explain?